temporal-serviceclient/src/main/java/io/temporal/internal/retryer/GrpcRetryerUtils.java

```diff
@@ -62,6 +66,10 @@ class GrpcRetryerUtils {
       case UNIMPLEMENTED:
         // never retry these codes
         return currentException;
+      case INTERNAL:
+        if (serverCapabilities.getInternalErrorDifferentiation()) {
+          return currentException;
+        }
```

Comment:
Would prefer a `break` here since fall through to deadline exceeded can be confusing to readers (ntm the deadline exceeded comment would not longer apply to just it). My normal rule of thumb for readability is to only allow case fall through if there are no statements. Your choice of course.

Reply:
Yeah, it totally makes sense.